src/requests/consumption.js

```diff
@@ -0,0 +1,377 @@
+			if (!category.label || !category.value) {
+				throw new Error('Each category must have a label and a value.')
+			}
+			return {
+				label: category.label,
+				value: category.value,
+				formattedName: formatCategoriesName(category.value),
+				externalId: category.value.replace(/_/g, '').toLowerCase(),
+			}
+		})
+
+		// Fetch existing categories by externalId
+		const existingCategories = await categoriesCollection
+			.find({ externalId: { $in: formattedCategories.map((cat) => cat.externalId) } })
+			.toArray()
+		const existingExternalIds = new Set(existingCategories.map((cat) => cat.externalId))
+
+		// Filter out categories that already exist
+		const newCategories = formattedCategories
+			.filter((cat) => !existingExternalIds.has(cat.externalId))
+			.map(({ formattedName, externalId, label }) => ({
+				createdBy: 'SYSTEM',
+				updatedBy: 'SYSTEM',
+				isDeleted: false,
+				isVisible: true,
+				status: 'active',
+				icon: '',
+				noOfProjects: 0,
+				name: formattedName,
+				externalId: externalId,
+				label: label,
+				createdAt: new Date(),
+				updatedAt: new Date(),
+			}))
+
+		// Insert only new categories
+		if (newCategories.length > 0) {
+			const result = await categoriesCollection.insertMany(newCategories)
+			newCategories.forEach((category, index) => {
+				category._id = result.insertedIds[index]
+				existingExternalIds.add(category.externalId)
+			})
+		}
+
+		// Map all categories to the response format
+		const processedCategories = formattedCategories.map((category) => {
+			const existing = existingCategories.find((cat) => cat.externalId === category.externalId)
+			if (existing) {
+				return {
+					_id: existing._id,
+					externalId: existing.externalId,
+					name: existing.name,
+				}
+			} else {
+				const newCategory = newCategories.find((cat) => cat.externalId === category.externalId)
+				return {
+					_id: newCategory._id,
+					externalId: newCategory.externalId,
+					name: newCategory.name,
+				}
+			}
+		})
+
+		return { success: true, categories: processedCategories }
+	} catch (error) {
+		console.error('Error in processCategories:', error.message)
+		return { success: false, error: `Failed to process categories: ${error.message}` }
+	}
+}
+
+/**
+ * Create Task
+ * @name createTasks
+ * @param {Object} tasks - task data
+ * @param {String} templateId - template Id
+ * @param {String} templateExternalId - template externaldId
+ * @param {String} parentId - parentId
+ * @returns {Object} - Response contains task data
+ */
+async function createTasks(tasks, templateId, templateExternalId, parentId = null) {
+	const result = { success: false, taskIds: [], externalIds: [], error: null }
+	try {
+		const taskCollection = mongoDb.collection(COLLECTIONS.TASKS)
+		const taskIds = []
+		const externalIds = []
+
+		for (const task of tasks) {
+			// Format the task data
+			const taskData = {
+				name: task.name,
+				description: task.name,
```

Comment:
Can we have `task?.description  ? task?.description  : ''`  In case of an adopter needs description for task.  @priyanka-TL

Reply:
@priyanka-TL  we dont have description key in scp. desription we are storing in scp as name, and also name = description as of now. Instaed of modifying each task unnecessory, we can add name and descriptio both same as of now, if they want this different, let scp need to handle that first